rfcs/192-fr-copy-receipt-performance.md

```diff
@@ -0,0 +1,177 @@
+- Feature Name: `fr_copy_receipt_performance`
+- Start Date: 2023-08-01
+- RFC PR: [fiskaltrust/middleware#192](https://github.com/fiskaltrust/middleware/pull/192)
+- Tracking Issue: [fiskaltrust/middleware#0000](https://github.com/fiskaltrust/middleware/issues/0000)
+
+# Summary
+
+Adding a `ftJournalFRCopyReceipts` table in the FR Queue where we store copy receipts.
+
+# Motivation
+
+The current implementation of the CopyReceipt is performing badly.
+Processing a CopyReceipt takes an increasing amount of time for each previous CopyReceipt in the queue.
+This inevitably leads to a point where the CopyReceipts are not processed in time.
+
+Currently when processing a CopyReceipt we have iterate over all previous CopyReceipts in the `ftJournalFR` table, deserialize the response and check if it is using the same `cbPreviousReceiptReference`.
+We then count all of those CopyReceipts and return that as the count.
+
+# Guide-level explanation
+
+For the PosCreator they are no implementation changes needed, the CopyReceipts still work like they did before.
+
+## New local queues
+
+New Queues don't need to take anything into account.
+They will work out of the box.
+
+## Updating local queues
+
+The new table will be created on start up after updating the queue.
+
+The data will be migrated from the `ftJournalFR` table to the new `ftJournalFRCopyReceipts` table.
+This will take a similar timeframe as it takes to process a CopyReceipt with the old implementation.
+
+## SignatureCloud
+
+The SignatureCloud will be updated as well.
```

Comment:
@Bluette-C-Riviere this is more of a certification question IMO. What would be the impact if we update all signaturcloud customers? Can you probably figure out how many of them are affected in if that makes sense?

Reply:
We have 4 customers/dealers using or having operating using the ChaineCloud

only 2 would need a new certificate, a third one will not be impacted because the certificate should be given mid-october.
The new certificate will be edited by the LNE after going through the change management process with have with them.